internals/api/idp_link.go

```diff
@@ -0,0 +1,35 @@
+package api
+
+import (
+	"time"
+
+	"github.com/secrethub/secrethub-go/pkg/oauthorizer"
+)
+
+type CreateIdentityProviderLinkGCPRequest struct {
+	RedirectURL       string `json:"redirect_url"`
+	AuthorizationCode string `json:"authorization_code"`
+}
+
+type IdentityProviderLinkType string
+
+const (
+	IdentityProviderLinkGCP IdentityProviderLinkType = "gcp"
+)
+
+type IdentityProviderLink struct {
```

Comment:
Maybe we can add some godoc explaining what an identity provider link is and when it is required?


It can use a lot of polishing, but I imagine something like this:
```suggestion
// IdentityProviderLink proves that you have access to a project within the identity provider.
// This link is required in order to create service accounts that use this identity provider.
// This is done so that other users cannot claim a SecretHub serivce account that uses the name of your GCP service account, which would keep you from using that name.
//
// For example, to create GCP service accounts, you first need to prove that you're a member of the GCP project in which you want to create the service account.
type IdentityProviderLink struct {
```

Reply:
Damn, that's difficult. Gave it a shot. But not sure whether it is very descriptive.